client/src/components/Messenger/ChatNavDrawer.Component.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { Fragment } from "react";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+import { Box, Button, Divider, Drawer, IconButton } from "@material-ui/core";
+import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
+import ChevronRightIcon from "@material-ui/icons/ChevronRight";
+import clsx from "clsx";
+import { useSelector, useDispatch } from "react-redux";
+import {
+  getClickContact,
+  getCloseNavDrawer,
+  getChatMessagesHistory,
+} from "../../store/actionCreator";
+import * as Components from "../Components";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    height: "100%",
+    width: "100%",
+    display: "flex",
+    flexDirection: "column",
+  },
+  drawerPaper: (props) => ({
+    width: props.drawerWidth,
+  }),
+  dividedPages: {
+    height: "100%",
+    display: "flex",
+    flexDirection: "column",
+  },
+  drawerFooter: {
+    width: "100%",
+    height: 60,
+    display: "flex",
+    alignItems: "center",
+    paddingTop: 3,
+    paddingBottom: 3,
+    justifyContent: "flex-end",
+  },
+  contactsListContainer: {
+    width: `calc(100% - 32px)`,
+    height: `calc(100% - 124px - 54px)`,
+    alignSelf: "center",
+    display: "flex",
+    overflow: "auto",
+    flexDirection: "column",
+    padding: "16px",
+    paddingTop: 0,
+  },
+  contactButtonClicked: {
+    background: "#e0e0e0",
+  },
+  hideDrawerCloseButton: {
+    display: "none",
+  },
+}));
+
+export default function ChatNavDrawerComponent(props) {
+  const classes = useStyles(props);
+  const theme = useTheme();
+
+  const drawerOpenStatus = useSelector((state) => state.get("navDrawerOpen"));
+  const hiddenChatChannel = useSelector((state) =>
+    state.get("hiddenChatChannel")
+  );
```

Comment:
Not every piece of state needs to be managed by Redux. If a piece of state is completely local to a component, it'd be better to just use `useState` for it.

Reply:
Yes, Thank you. Because I just learned Redux last Friday, and I thought it's much friendly than useState. So I keep the code as the same way.